api/dataprocessor.go

```diff
@@ -466,7 +467,7 @@ func (s *Server) getSeriesCachedStore(ctx *requestContext, until uint32) []chunk
 	// the request cannot completely be served from cache, it will require cassandra involvement
 	if !cacheRes.Complete {
 		if cacheRes.From != cacheRes.Until {
-			storeIterGens, err := s.BackendStore.Search(key, cacheRes.From, cacheRes.Until)
+			storeIterGens, err := s.BackendStore.Search(key, ctx.TTL, cacheRes.From, cacheRes.Until)
```

Comment:
Right, good catch. I think that means I'll need to modify `models.Req` and add the `TTL` there. Then that `TTL` attribute cat get set in `alignRequests()` when the aggregation is selected.